packages/palette/src/elements/Button/Button.tsx

```diff
@@ -172,214 +168,219 @@ export const getStylesForVariant = (variant: ButtonVariant) => {
-                color: ${colors.black100};
-
-                @media ${themeProps.mediaQueries.hover} {
-                  &:hover {
-                    background-color: ${colors.white100};
-                    border-color: ${colors.black100};
-                    color: ${colors.black100};
-                  }
+              }
+            `
+        }};
+      `
+    case "secondaryOutline":
+      return css`
+        ${props => {
+          const { colors } = props.theme
+          return `
+              background-color: ${colors.white100};
+              border-color: ${colors.black10};
+              color: ${colors.black100};
+
+              @media ${themeProps.mediaQueries.hover} {
+                &:hover {
+                  background-color: ${colors.white100};
+                  border-color: ${colors.black100};
+                  color: ${colors.black100};
                 }
-              `
-          }};
-        `
-      case "noOutline":
-        return css`
-          ${props => {
-            const { colors } = props.theme
-            return `
-                background-color: transparent;
-                border-color: transparent;
-                color: ${colors.black100};
-              `
-          }};
-        `
-      default:
-    }
+              }
+            `
+        }};
+      `
+    case "noOutline":
+      return css`
+        ${props => {
+          const { colors } = props.theme
+          return `
+              background-color: transparent;
+              border-color: transparent;
+              color: ${colors.black100};
+            `
+        }};
+      `
+    default:
   }
+}
 
-  render() {
-    const buttonProps = {
-      ...this.props,
-      ...this.getSize(),
-      buttonSize: this.props.size,
-      variantStyles: this.getVariant(),
-    }
-
-    return <ButtonBase {...buttonProps}>{this.props.children}</ButtonBase>
-  }
+/** A button with various size and color settings */
+export const Button: React.FC<ButtonProps> = ({
+  size = defaultSize,
+  variant = defaultVariant,
+  children,
+  ...rest
+}) => {
+  return (
+    <ButtonBase
+      {...rest}
+      {...getSize({ inline: rest.inline, size })}
+      buttonSize={size}
+      variantStyles={getVariant({ variant })}
+    >
+      {children}
+    </ButtonBase>
+  )
 }
 
 /** A base from which various button implementations can compose from */
-export class ButtonBase extends Component<ButtonBaseProps & SansProps> {
-  static defaultProps = {
-    border: 1,
-    borderRadius: 3,
-  }
-
-  onClick = event => {
-    if (!this.props.loading && this.props.onClick) {
-      this.props.onClick(event)
+export const ButtonBase: React.FC<ButtonBaseProps & SansProps> = ({
+  children,
+  loading,
+  disabled,
+  color,
+  size,
+  longestText,
+  weight,
+  className,
+  onClick,
+  ...rest
+}) => {
+  const loadingClass = loading ? "loading" : ""
+  const disabledClass = disabled ? "disabled" : ""
```

Comment:
I don't suppose there's any real reason these are classes? Do we do some kind of weird DOM manipulation somewhere where we have to add/remove classes?

Reply:
I'm a little worried to remove as I'm sure this was added for a reason -- _but_ I'm also thinking that whatever relied on adding / removing classes in this way is super old and thus fine to break. Better to just clean up confusing primitive code and wait for an issue to arise.